src/libcore/time.rs

```diff
@@ -224,6 +224,49 @@ impl Duration {
         }
     }
 
+    /// Creates a new `Duration` that spans no time.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(duration_zero)]
+    /// use std::time::Duration;
+    ///
+    /// let duration = Duration::zero();
+    /// assert!(duration.is_zero());
+    ///
+    /// const IMMEDIATELY: Duration = Duration::zero();
+    /// assert!(IMMEDIATELY.is_zero());
```

Comment:
I don't think the `const` example adds a lot here. I would rather add an `assert_eq!(duration.as_nanos(), 0)` or something like that.